idaes/power_generation/carbon_capture/mea_solvent_system/property_package/liquid_prop.py

```diff
@@ -0,0 +1,1282 @@
+                return -84000
+        self.habs = Expression(rule = rule_habs,
+                                doc="CO2 heat of absorption [J/mol]")
+
+    def _hvap(self):
+        '''
+        Heat of vaporization of H2O
+        '''
+        def rule_hvap(blk):
+            c  = blk._params.heat_vaporization_param
+            T  = blk.temperature
+            Tc = blk._params.temperature_crit['H2O']
+            return 1000*c[1]*(1-T/Tc)**(c[2]+c[3]*T/Tc+c[4]*(T/Tc)**2)
+        self.hvap = Expression(rule = rule_hvap,
+                                doc="Heat of vaporization of H2O [J/mol]")
+
+    def _henry_N2O_analogy(self):
+        '''
+        Henry's constant N2O Analogy Jiru et.al (2012)
+        '''
+        def rule_henry_N2O_analogy(blk):
+            T      = blk.temperature
+            t      = blk.temperature - 273.15
+            wt_MEA = blk.mass_frac1['MEA']
+            wt_H2O = blk.mass_frac1['H2O']
+            H_N2O_MEA = 2.448e5*exp(-1348/T)
+            H_CO2_H2O = 3.52e6*exp(-2113/T)
+            H_N2O_H2O = 8.449e6*exp(-2283/T)
+            H_CO2_MEA = H_N2O_MEA*(H_CO2_H2O/H_N2O_H2O)
+            lwm = 1.70981 + 0.03972*t - 4.3e-4*t**2 - 2.20377*wt_H2O
+
+            return (exp(wt_MEA*log(H_CO2_MEA) +wt_H2O*log(H_CO2_H2O)
+                   + wt_MEA*wt_H2O*lwm))
+
+        self.henry_N2O_analogy = Expression(rule = rule_henry_N2O_analogy,
+                                doc="Henry's constant [Pa.m^3/mol]")
+
+    #--------------------------------------------------------------------------
+    def get_material_flow_terms(b, p, j):
+        return b.flow_mol_comp[j]
+
+    def get_enthalpy_flow_terms(b, p):
+        return b.enth_mol_mean
+
+    def get_material_density_terms(b, p, j):
+        return b.conc_comp[j]
+
+    def get_energy_density_terms(b, p):
+        return b.enth_mol_liq_density
+
+    def define_state_vars(b):
+        return {"flow_mol": b.flow_mol,
+                "temperature": b.temperature,
+                "pressure": b.pressure,
+                "mole_frac": b.mole_frac}
+
+    def get_material_flow_basis(b):
+        return MaterialFlowBasis.molar
+
+    def model_check(blk):
+        """
+        Model checks for property block
+        """
+        # Check temperature bounds
+        if value(blk.temperature) < blk.temperature.lb:
+            _log.error('{} Temperature set below lower bound.'
+                       .format(blk.name))
+        if value(blk.temperature) > blk.temperature.ub:
+            _log.error('{} Temperature set above upper bound.'
+                       .format(blk.name))
+
+        # Check pressure bounds
+        if value(blk.pressure) < blk.pressure.lb:
+            _log.error('{} Pressure set below lower bound.'.format(blk.name))
+        if value(blk.pressure) > blk.pressure.ub:
+            _log.error('{} Pressure set above upper bound.'.format(blk.name))
+
+
+def evaluate_property_package():
```

Comment:
Why is this part of the property package file? This looks like it should be part of the tests, or maybe an example.

Also, many of these imports are duplicated of those at the beginning of the file.

Reply:
This has been removed.